docs/baremetal-operator-on-kubernetes-cluster.md

```diff
@@ -0,0 +1,179 @@
+# Running Baremetal-operator on kubernetes cluster
+
+## Description
+
+This document outlines the process of provisioning completely within
+the cluster; starting with spinning the components
+**Baremetal-operator and Ironic** over a
+**On-Premises Kubernetes Cluster**, till the **provisioning of baremetal
+machines** using only cluster resources.
+
+## Prerequisites
+
+Fully ready Kubernetes cluster (version >= 1.17)
+
+## Spinning
+
+* ### Clone the **Baremetal-operator**
+
+```console
+> git clone https://github.com/metal3-io/baremetal-operator.git
+> cd baremetal-operator
+```
+
+* ### Start **Ironic**
+
+We will be applying the ironic first as we need the cluster IP
+of ironic pod to configure it in the config-maps of Baremetal-operator for
+inter-pod communication.
```

Comment:
this should be done after the configmap update. Otherwise Ironic will start with wrong values and not pick on the update.

Reply:
I didn't quite get you, baremetal-op config maps must be updated with ironic ClusterIP; and for that ironic should be running first, isn't it?
I am not much aware of others ways to do it. I found this way (document) to be successfully running and provisioning.
As I couldn't find any documentation for running the metal3-> baremetal-op completely in an on-prem cluster env. 
I will be happy to take suggestions.

Reply:
ok, sorry for being unclear. When you deploy ironic, it also uses a configmap that is in ironic-deployment/default. In your case it probably worked because you anyways have a eth2 interface (default there) and would use pretty much the same values. However, if you were not using those default values and did not update that configmap, ironic would not start. Be careful that this configmap is not the Baremetal Operator configmap that you describe later. You need to do the same update for the ironic configmap.